cmd/gateway-gcs.go

```diff
@@ -684,30 +692,24 @@ func (l *gcsGateway) PutObject(bucket string, key string, size int64, data io.Re
 
 	w.ContentType = metadata["content-type"]
 	w.ContentEncoding = metadata["content-encoding"]
+	w.Metadata = metadata
```

Comment:
The reason is that only after w.Close() we can validate the sha256sum. i.e after a flush.